src/connection.ts

```diff
@@ -39,7 +39,10 @@ export const SFDX_HTTP_HEADERS = {
 };
 
 export const DNS_ERROR_NAME = 'Domain Not Found';
-const DNS_TIMEOUT = Math.max(3, new Env().getNumber('SFDX_DNS_TIMEOUT') || 3);
+// Timeout for DNS lookup polling defaults to 3 seconds and should always be at least 3 seconds
+const DNS_TIMEOUT = Math.max(3, new Env().getNumber('SFDX_DNS_TIMEOUT', 3) as number);
```

Comment:
These are generic envvars. What are your thoughts about also putting these in myDomainResolver:98 so all DNS checks follow it? Otherwise, maybe they should be `SFDX_CONNECTION_DNS_XX`?